src/main/common/filter.c

```diff
@@ -122,16 +128,25 @@ void biquadFilterInit(biquadFilter_t *filter, uint16_t filterFreq, uint32_t refr
     filter->a1 = a1 / a0;
     filter->a2 = a2 / a0;
 
-    // zero initial samples
-    filter->d1 = filter->d2 = 0;
+    filter->filterFreq = filterFreq;
+    filter->refreshRate = refreshRate;
+    filter->running = false;
 }
 
 // Computes a biquad_t filter on a sample
 float biquadFilterApply(biquadFilter_t *filter, float input)
 {
-    const float result = filter->b0 * input + filter->d1;
-    filter->d1 = filter->b1 * input - filter->a1 * result + filter->d2;
-    filter->d2 = filter->b2 * input - filter->a2 * result;
+    float result;
+    if (!filter->running) {
```

Comment:
> Compared to five floating multiplications I still believe the additional cost of a test is negligible

1. It's not negligible. The test operation can cause a pipeline stall and, in the worst case, an instruction cache miss, so the seemingly innocuous operation can be expensive.
2. When dealing with speed optimisations things are often counterintuitive, so what we believe is only a first approximation. What is important is what is measured.
3. Even if it was negligible, it's still unnecessary. As I said before, the test should be executed at a higher level.

Note that the filter code is shared with betaflight and is used for 32kHz looptimes. With 32kHz looptimes you have at most 31.5 microseconds per loop - so every instruction counts.

As far as I am aware, the servo code is not executed unless required. If it is executed then it's a bug and needs to be corrected.